lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -1175,11 +1180,11 @@ checkpointId cp = (checkpointWalletId cp, checkpointSlot cp)
 
 -- | Functions for saving/loading the wallet's address discovery state into
 -- SQLite.
-class PersistState s where
+class PersistState s k where
     -- | Store the state for a checkpoint.
-    insertState :: (W.WalletId, W.SlotId) -> s -> SqlPersistT IO ()
+    insertState :: (W.WalletId, W.SlotId) -> s -> AddressScheme k-> SqlPersistT IO ()
     -- | Load the state for a checkpoint.
-    selectState :: (W.WalletId, W.SlotId) -> SqlPersistT IO (Maybe s)
+    selectState :: (W.WalletId, W.SlotId) -> AddressScheme k -> SqlPersistT IO (Maybe s)
```

Comment:
Arguably looks a bit weird to take `AddressScheme k`, but I think it is fine, and can be improved upon in the future.